sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_client_connection.py

```diff
@@ -131,6 +132,7 @@ def __init__(
             The default consistency policy for client operations.
 
         """
+        self.client_id = str(uuid.uuid4())
```

Comment:
Is this expected to be updated at all? Or will remain constant throughout the client lifetime?
(just thinking in terms of threadsafety)

Reply:
Expected and desired to be constant throughout client lifetime so effectively immutable.